src/pybind/mgr/dashboard/frontend/src/app/ceph/new-dashboard/dashboard/dashboard.component.ts

```diff
@@ -37,4 +85,50 @@ export class DashboardComponent implements OnInit {
         version[0] + ' ' + version.slice(2, version.length).join(' ');
     });
   }
+
+  getDf() {
+    this.clusterService.getCapacity().subscribe((data: any) => {
+      this.dfData = data;
+    });
+  }
+
+  prepareRawUsage(chart: Record<string, any>, data: Record<string, any>) {
+    const nearFullRatioPercent = this.osdSettings.nearfull_ratio * 100;
+    const fullRatioPercent = this.osdSettings.full_ratio * 100;
+    const percentAvailable = this.calcPercentage(
+      data.stats.total_bytes - data.stats.total_used_raw_bytes,
+      data.stats.total_bytes
+    );
+    const percentUsed = this.calcPercentage(
+      data.stats.total_used_raw_bytes,
+      data.stats.total_bytes
+    );
+    if (percentUsed >= fullRatioPercent) {
+      this.color = 'chart-color-red';
+    } else if (percentUsed >= nearFullRatioPercent) {
+      this.color = 'chart-color-yellow';
+    } else {
+      this.color = 'chart-color-blue';
+    }
+
+    chart.dataset[0].data = [
+      Math.round(nearFullRatioPercent),
+      Math.round(Math.abs(nearFullRatioPercent - fullRatioPercent)),
+      Math.round(100 - fullRatioPercent)
+    ];
+
+    chart.dataset[1].data = [percentUsed, percentAvailable];
+    chart.dataset[1].backgroundColor[0] = this.cssHelper.propertyValue(this.color);
+
+    chart.dataset[0].label = [
+      `${percentUsed}%\nof ${this.dimlessBinary.transform(data.stats.total_bytes)}`
+    ];
+  }
```

Comment:
This is exposing externally the internal way chartjs works with data and styles. I know this is how we did it originally (but this code is 'ancient' and we've learnt a lot since then). So what about decoupling those? Data takes 1 path, and styling another one.

Potentially I'd expect to see something like this. The only actual data the consumer of this should provide is the current data and the total/max (the pie component should internally calculate the remaining part, thresholds and so):
```html
<cd-dashboard-pie [data]="{max: data.stats.total_bytes, current: data.stats.total_used_raw_bytes}" thresholdHigh=...
```
For the `[data]` you'll have to define an interface (`IData`).